src/libsystemd/sd-device/sd-device.c

```diff
@@ -1812,7 +1812,7 @@ static int device_sysattrs_read_all_internal(sd_device *device, const char *subd
                 if (lstat(path, &statbuf) != 0)
                         continue;
 
-                if (!(statbuf.st_mode & S_IRUSR))
+                if ((statbuf.st_mode & (S_IRUSR | S_IWUSR)) == 0)
```

Comment:
`FLAGS_SET`?

Reply:
No. `FLAGS_SET()` returns true only when all flags are set. So,
`FLAGS_SET(statbuf.st_mode, S_IRUSR | S_IWUSR)` is true only when the file is rw.